external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -236,6 +257,35 @@ private boolean poll() {
         return poll;
     }
 
+    private void refreshPartitionIfNeeded() {
+        if (!manualPartitionAssignment || !partitionRefreshTimer.isExpiredResetOnTrue()) return;
+        doRefreshPartitions();
+    }
+
+    private void doRefreshPartitions() {
+        KafkaSpoutStreamsNamedTopics streams = KafkaSpoutStreamsNamedTopics.class.cast(kafkaSpoutStreams);
```

Comment:
1. Wildcard topics support are added.
2. class.cast is preferred because it throws an exception which better explains the failure rather than NPE.